posthog/api/routing.py

```diff
@@ -28,10 +31,10 @@ def __init__(self, *args, **kwargs):
         self.trailing_slash = r"/?"
 
 
-class StructuredViewSetMixin(_GenericViewSet):
+class TeamAndOrgViewSetMixin(_GenericViewSet):
     # This flag disables nested routing handling, reverting to the old request.user.team behavior
     # Allows for a smoother transition from the old flat API structure to the newer nested one
-    legacy_team_compatibility: bool = False
+    derive_current_team_from_user: bool = False
```

Comment:
Maybe `derive_current_team_from_user_only`, because this doesn't need to be True for deriving the current team from the user – it's always possible with the explicit special value `@current`. This is only for leaving those legacy endpoints around

Reply:
Ah good call